src/trs/rewrite.rs

```diff
@@ -309,6 +309,165 @@ impl TRS {
             Ok(trs)
         }
     }
+    /// Selects a rule from the TRS at random, finds all differences in the LHS and RHS,
+    /// and makes rules from those differences and inserts them back into copies of the TRS imediately after the background.
+    pub fn local_difference_vec<R: Rng>(&self, rng: &mut R) -> Result<Vec<TRS>, SampleError> {
```

Comment:
Oops, noticed a couple things I missed the first time.

Could you please add tests to the public-facing functions?